zcash_client_backend/src/serialization/shardtree.rs

```diff
@@ -43,6 +44,10 @@ pub fn write_shard<H: HashSer, W: Write>(writer: &mut W, tree: &PrunableTree<H>)
                 writer.write_u8(NIL_TAG)?;
                 Ok(())
             }
+            Node::Pruned => {
+                writer.write_u8(PRUNED_TAG)?;
+                Ok(())
```

Comment:
This will not be readable by prior implementations of `read_shard_v1`. Should the version number be bumped, or would that just cause more problems?

Reply:
I think this is fine; we can treat version changes that are backwards-compatible as part of the same version.